warmrestart/warmRestartAssist.cpp

```diff
@@ -173,7 +174,7 @@ void AppRestartAssist::insertToMap(string key, vector<FieldValueTuple> fvVector,
     else if (found != appTableCacheMap.end())
     {
         // check only the original vector range (exclude cache-state field/value)
-        if(!equal(fvVector.begin(), fvVector.end(), found->second.begin()))
+        if(! contains(found->second, fvVector))
```

Comment:
Do we want to be exact match instead of "contains"?  We may need sort both vectors (exclude the "state") first. 
The vector (exclude the "state") should be exactly the same as before, even if the new vector was a subset, we should consider it as a "change".

Reply:
@zhenggen-xu in principle, I agree with you. But I lost the repeating system that I had last week and therefore I feel comfortable check in this tested code. Then I don't have objection for the version you proposed. At this point, neighbor sync is the only user of this class, and it exactly expects 2 key value pairs that doesn't overlap. So contains works the same as exact match and exclude the state.

I am sorry that I didn't investigate more over the weekend. I thought this class was a generic class used by many clients. That is why I made this change which doesn't have expectation of the location of the state. and doesn't have expectation of the location of the key/value set needs to be matched. So even if the matching set is a subset of the full list at random places, my change would work.

Reply:
@yxieca You code should work for the cases in the current SONiC as the FVs in the vector are determined.   I don't see this will break anything in the current SONiC code, so I am fine to use this change as you already tested in the field and feel more comfortable.